test/distance.jl

```diff
@@ -44,6 +46,63 @@
             @test @inferred(center(z)) == center(g, distmx2) == [2]
         end
     end
+
+    @testset "$(typeof(g))" for g in test_generic_graphs(a3)
```

Comment:
The name of this testset is a bit cryptic to me. Could you elaborate it so that it is more immediately understandable if a test failure is reported?